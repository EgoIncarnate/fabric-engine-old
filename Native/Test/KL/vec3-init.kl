struct Vec3
{
  Scalar x;
  Scalar y;
  Scalar z;
};

function entry()
{
  var Vec3 a;
  a.x = 1; report a.x;
  a.y = 2; report a.y;
  a.z = 3; report a.z;

  var Vec3 b = a;
  report b.x;
  report b.y;
  report b.z;
}
