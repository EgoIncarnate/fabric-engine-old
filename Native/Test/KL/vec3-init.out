GlobalList
  StructDecl( 'Vec3' )
    StructDeclMember( 'x', Scalar )
    StructDeclMember( 'y', Scalar )
    StructDeclMember( 'z', Scalar )
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          VarDecl( 'a', Vec3 )
        ExprStatement
          AssignOp(ASSIGN_OP)
            StructMemberOp( 'x' )
              Var('a')
            ConstSize( 1 )
        Report
          StructMemberOp( 'x' )
            Var('a')
        ExprStatement
          AssignOp(ASSIGN_OP)
            StructMemberOp( 'y' )
              Var('a')
            ConstSize( 2 )
        Report
          StructMemberOp( 'y' )
            Var('a')
        ExprStatement
          AssignOp(ASSIGN_OP)
            StructMemberOp( 'z' )
              Var('a')
            ConstSize( 3 )
        Report
          StructMemberOp( 'z' )
            Var('a')
        StatementList
          AssignedVarDecl( VarDecl( 'b', Vec3 ), Var('a') )
            Var('a')
        Report
          StructMemberOp( 'x' )
            Var('b')
        Report
          StructMemberOp( 'y' )
            Var('b')
        Report
          StructMemberOp( 'z' )
            Var('b')
1
2
3
1
2
3
